src/components/common/ConnectWallet/MPCLogin.tsx

```diff
@@ -8,18 +8,41 @@ import InfoIcon from '@/public/images/notifications/info.svg'
 
 import css from './styles.module.css'
 import useWallet from '@/hooks/wallets/useWallet'
-import { useCurrentChain } from '@/hooks/useChains'
-import chains from '@/config/chains'
+import useChains, { useCurrentChain } from '@/hooks/useChains'
+import { isSocialWalletEnabled } from '@/hooks/wallets/wallets'
+import { ONBOARD_MPC_MODULE_LABEL } from '@/services/mpc/module'
+import { CGW_NAMES } from '@/hooks/wallets/consts'
+import { type ChainInfo } from '@safe-global/safe-gateway-typescript-sdk'
 
-const MPCLogin = ({ onLogin }: { onLogin?: () => void }) => {
+export const _getSupportedChains = (chains: ChainInfo[]) => {
+  return chains.reduce((result: string[], currentChain) => {
+    if (CGW_NAMES.SOCIAL_LOGIN && !currentChain.disabledWallets.includes(CGW_NAMES.SOCIAL_LOGIN)) {
```

Comment:
Do you know why we type the mapped type of `CGW_NAMES` as `string | undefined`?

Can it really be undefined? 
Should we refactor it as 
```ts
export const CGW_NAMES = {
  [WALLET_KEYS.INJECTED]: 'detectedwallet',
  [WALLET_KEYS.WALLETCONNECT_V2]: 'walletConnect_v2',
  [WALLET_KEYS.COINBASE]: 'coinbase',
  [WALLET_KEYS.PAIRING]: 'safeMobile',
  [WALLET_KEYS.SOCIAL]: 'socialLogin',
  [WALLET_KEYS.LEDGER]: 'ledger',
  [WALLET_KEYS.TREZOR]: 'trezor',
  [WALLET_KEYS.KEYSTONE]: 'keystone',
} as const
```

Or am I missing something about this mapping?

Reply:
I was also wondering about that. The only reason I can see is that the `WALLET_MODULES` and `CGW_NAMES` should be coupled but currently are not so it could happen that we try to access a key in `CGW_NAMES` that doesn't exist.